src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,133 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    this._client = this.createClient();
+    this._noAuthClient = this.createClient({ auth: false });
+
+    return this;
+  }
+
+  /**
+   * callWithRequest
+   *
+   * Performs a call to ES, passing through whitelisted headers in the request
+   *
+   * @param {Object|undefined} req - The request object
+   * @param {string} endpoint
+   * @param {Object} clientParams
+   * @param {Object} options
+   * @param {boolean} options.wrap401Errors
```

Comment:
@tylersmalley - I just ran into this JSDOC - https://code.google.com/archive/p/jsdoc-toolkit/wikis/TagParam.wiki
See the *Parameters With Properties* section
Apparently 
```
@param {Object} options
@param {boolean} options.wrap401Errors
```
is the legit way to say you expect the options Object to have that particular boolean parameter, without having to use typedef.   Maybe you already knew that, but in case you didn't, wanted to share.  :)